Makefile.pre.in

```diff
@@ -440,7 +440,15 @@ DTRACE_DEPS = \
 
 # Default target
 all:		@DEF_MAKE_ALL_RULE@
-build_all:	$(BUILDPYTHON) oldsharedmods sharedmods gdbhooks Programs/_testembed python-config
+build_all:	check-clean-src $(BUILDPYTHON) oldsharedmods sharedmods gdbhooks \
+		Programs/_testembed python-config
+
+# Check that the source is clean when building out of source.
+check-clean-src:
+	@if test -n "$(VPATH)" -a -f "$(srcdir)/Programs/python.o"; then \
+		echo "Error: The source directory is not clean" ; \
```

Comment:
The error message is not easy to understand. I propose:

```
		echo "Error: The source directory ($(srcdir)) is not clean" ; \
		echo "Building Python out of the source tree ($(abs_builddir)) requires a clean source tree ($(abs_srcdir)" ; \
		echo "Try to run: make -C \"$(srcdir)\" clean" ; \
```

At least, the proposed command worked for me ;-)

Reply:
Ok